Feature/profiles

Naenyn opened the pull request:
This is an initial attempt at adding profiles for uPortal 5 and 6 with their associated, version-specific configurations.

Naenyn committed:
Parent pom WIP

Naenyn committed:
Parent pom WIP

Naenyn committed:
First pass at uP5/6 profiles.

bjagg commented on pom.xml:
```diff
@@ -108,10 +106,194 @@
+        <dependency>
+          <groupId>org.apache.portals.pluto</groupId>
+          <artifactId>pluto-taglib</artifactId>
+          <!-- does uP6 need v3.x of this? -->
```
Use the version uPortal 5.17 uses

Naenyn replied:
Got it!

bjagg commented on pom.xml:
```diff
@@ -108,10 +106,194 @@
+          <scope>provided</scope>
+        </dependency>
+
+        <!-- Liferay's port of Spring portlets (needing for Spring 5+) -->
```
Remove for uP5

Naenyn replied:
This line is contained w/in the uP6 profile. Check again?

bjagg reviewed (changes requested):
Looks good. Liferay is not used for portlets in uP5, so those should not be in the uP5 profile. Is there a uP6 profile?

Naenyn commented:
> Is there a uP6 profile?

Yep, and you appear to have found it given your second PR comment! I'll get those changes in.

Naenyn committed:
Moved pluto-taglib back outside of profiles.

Both uP5 and uP6 use the same version, so no need to keep it within the
profiles.

bjagg commented:
No longer needed with our new roadmap.
